### PR TITLE
Implement drag'n'drop for observation list

### PR DESCRIPTION
**Pull request by hugo-vrijswijk**
Depends on https://github.com/gemini-hlsw/lucuma-react/pull/628 and https://github.com/gemini-hlsw/explore/pull/3346

### CODE REVIEW

File: explore/src/main/scala/explore/common/GroupQueries.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+object GroupQueries:
+
+  /**
+   * @param groupId
+   *   Group to move
+   * @param parentGroup
+   *   New parent group, `None` to move to top level
+   * @param parentGroupIndex
+   *   New index in parent group, `None` to leave position unchanged
```
Comment (toddburnside) [inline explore/src/main/scala/explore/common/GroupQueries.scala:25]: What does it mean to leave the index `unchanged` if it is a new parent group?

**Review by toddburnside — APPROVED** [review]
LGTM. Hopefully Prime React will release a new version soon. Andy will be very excited. 😄

File: explore/src/main/scala/explore/common/GroupQueries.scala (hunk 1 above)
Comment (hugo-vrijswijk) [reply explore/src/main/scala/explore/common/GroupQueries.scala:25]: 'Unchanged' is definitely the wrong word. It means "let the odb figure out the index". Though in reality this is always filled (primereact-tree gives always gives the index)

**Comment by hugo-vrijswijk** [comment]
> LGTM. Hopefully Prime React will release a new version soon. Andy will be very excited. 😄

There is a new release. But the CI disagrees 🙃

**Comment by toddburnside** [comment]
> There is a new release. But the CI disagrees 🙃

Dang computers.

File: explore/src/main/scala/explore/observationtree/obsTree.scala [hunk 2]
```diff
@@ -8,40 +8,59 @@ import cats.Order.*
...
+import explore.data.tree.{Tree => IndexTree}
+import explore.model.*
+import explore.model.syntax.all.*
+import lucuma.core.model.Group
 import lucuma.core.model.Observation
 import lucuma.react.primereact.Tree
 import lucuma.react.primereact.Tree.Node
+import monocle.Iso
 
 enum ObsNode derives Eq:
```
Comment (cquiroz) [inline explore/src/main/scala/explore/observationtree/obsTree.scala:24]: Why an `enum`? It seems this would be more an `ADT`

**Review by cquiroz — APPROVED** [review]
Thanks, LGTM

File: explore/src/main/scala/explore/observationtree/obsTree.scala (hunk 2 above)
Comment (hugo-vrijswijk) [reply explore/src/main/scala/explore/observationtree/obsTree.scala:24]: Aren't enums with params basically ADT's?

Comment (toddburnside) [reply explore/src/main/scala/explore/observationtree/obsTree.scala:24]: Yes

**Pull request merged**